### ISSUE TITLE
eyeD3 doesn't save images with UTF8 encoding when description is empty

### ISSUE THREAD

**Issue by fcastilloec** [nicfit/eyeD3#200]
I tried saving an image using the python module to ID3_V2_4 using UTF8, all the fields are saved with UTF8 except for the image field. The image field is saved using ISO-8859-1
Here's a sample code to reproduce the problem:
```python
FRONT_COVER = eyed3.id3.frames.ImageFrame.FRONT_COVER

audiofile = eyed3.load(file_name)
IMAGES = audiofile.tag.images
IMAGES.set(FRONT_COVER, image_data, mime_type, u"")

audiofile.tag.save(encoding='utf-8', version=eyed3.id3.ID3_V2_4)
```

That code saves the APIC tag (picture) with ISO-8859-1, but if the `set` funciton is as follow:
`IMAGES.set(FRONT_COVER, image_data, mime_type, u"description")` then the image field is indeed saved with UTF-8

I should say that I was using Ubuntu 16.04 which came with eyeD3 v0.6.18, that version (using python2) didn't have this bug. I've just upgraded to Ubuntu 18.04 and found this bug when updating my scripts to the new version

**Comment by nicfit** [comment]
Please upgrade eyeD3 to the lastest version (eyeD3 0.6 is 6+ years out of date) and attempt to reproduce.

**Comment by fcastilloec** [comment]
I'm sorry for not being clear in my previous post. The issue happens with the latest version of eyeD3 (`v0.8.4`), the one found in the Ubuntu 18.04 repositories. On my previous comment, I wasn't clear enough. This issue didn't happen on v0.6.18 which was that last version I was using before upgrading to Ubuntu 18.04. Now that I've installed the latest Ubuntu (fresh install) and using the latest version of eyeD3, I'm trying to migrate all my scripts (as I mentioned in the original post) from v0.6.18 to v0.8.4 and I'm encountering this issue. Also, there's no migration guide anywhere for the change of API.

As you can recognize, the API used in the example code I originally posted is using the latest version of eyeD3. That code wouldn't have worked on version 0.6.18

**Comment by reardenlife** [comment]
It looks like I am having the same issue.
v0.8.9

**Comment by nicfit** [comment]
Here is my version of the test script:
```python
import eyed3.id3

FRONT_COVER = eyed3.id3.frames.ImageFrame.FRONT_COVER

audiofile = eyed3.load("test.id3")
IMAGES = audiofile.tag.images

with open("./FRONT_COVER.jpg", "rb") as fp:
    IMAGES.set(FRONT_COVER, fp.read(), "img/jpg", u"")

audiofile.tag.save(encoding='utf-8', version=eyed3.id3.ID3_V2_4)
```
And here are the results with 0.8.9:
```
$ eyeD3 test.mp3 
/home/travis/devel/eyeD3/git/test.mp3                                                                          [ 91.86 KB ]
----------------------------------------------------------------------------------------------------------------------------
Time: 00:05	MPEG1, Layer III	[ 128 kb/s @ 44100 Hz - Joint stereo ]
--------------------------------------------------------------------------------------------------------------------------------
ID3 v2.4:
title: 
artist: 
album: foo
track: 		
FRONT_COVER Image: [Size: 18113 bytes] [Type: img/jpg]
Description: 
```
Likewise, `eyeD3 --write-images . test.mp3` extracts the image successfully.
Let me know how to reproduce.

**Comment by fcastilloec** [comment]
I've updated to v0.8.9 and using your script, I still see this problem. To be able to check for the problem, you'll need another tool, I use `mid3v2` (part of `python-mutagen` in Ubuntu) or [kid3](https://kid3.sourceforge.io/), eyeD3 doesn't display the encoding of each of the fields, just the content of them.

I'm going to try to explain a little better using `mid3v2`. Here's the output of what my mp3 files looks like (it's also ID3v2.3 which is not shown):
```
$ mid3v2 --list-raw test.mp3
Raw IDv2 tag info for test.mp3
TALB(encoding=<Encoding.UTF16: 1>, text=[u'test'])
TPE1(encoding=<Encoding.UTF16: 1>, text=[u'test'])
APIC(encoding=<Encoding.LATIN1: 0>, mime=u'image/jpeg', type=<PictureType.COVER_FRONT: 3>, desc=u'', data=...)
TIT2(encoding=<Encoding.UTF16: 1>, text=[u'test'])
```

Here's an even simpler test code which simply loads an mp3 and saves it using UTF8 encoding and `ID3V2.4`:
```python
import eyed3.id3

audiofile = eyed3.load("test.mp3")
audiofile.tag.save(encoding='utf-8', version=eyed3.id3.ID3_V2_4)
```
The output of this code with v0.8.9 is:
```
$ mid3v2 --list-raw test.mp3
Raw IDv2 tag info for test.mp3
TIT2(encoding=<Encoding.UTF8: 3>, text=[u'test'])
TPE1(encoding=<Encoding.UTF8: 3>, text=[u'test'])
APIC(encoding=<Encoding.LATIN1: 0>, mime=u'image/jpeg', type=<PictureType.COVER_FRONT: 3>, desc=u'', data=...)
TALB(encoding=<Encoding.UTF8: 3>, text=[u'test'])
```

Notice that the APIC field (images) didn't change it's encoding at all. In previous versions of `eyeD3`, particularly v0.6.18, all fields will be UTF8, including APIC field.

**EDIT**: Any MP3 file will do for this test. The most important thing to reproduce is that the **description of the APIC field is always blank/empty**. If there's a string there, anything, then the APIC field does show as UTF8 with any of the scripts.

**Comment by nicfit** [comment]
I get it now. Fixed in b573bb7 .
